src/main/java/com/google/devtools/build/lib/bazel/commands/VendorCommand.java

```diff
@@ -309,58 +326,71 @@ private ImmutableSet<RepositoryName> collectReposFromTargets(
    */
   private void vendor(
       CommandEnvironment env,
-      PathFragment vendorDirectory,
+      Path vendorDirectory,
       ImmutableList<RepositoryName> reposToVendor)
-      throws IOException {
-    Path vendorPath =
-        vendorDirectory.isAbsolute()
-            ? env.getRuntime().getFileSystem().getPath(vendorDirectory)
-            : env.getWorkspace().getRelative(vendorDirectory);
-    Path externalPath =
-        env.getDirectories()
-            .getOutputBase()
-            .getRelative(LabelConstants.EXTERNAL_REPOSITORY_LOCATION);
+      throws IOException, InterruptedException {
+    VendorUtil vendorUtil = new VendorUtil(vendorDirectory);
 
-    if (!vendorPath.exists()) {
-      vendorPath.createDirectory();
-    }
+    // 1. Vendor registry files
```

Comment:
Does this also clean up unused registry files?

Reply:
Nope, we also don't clean up unused vendored repo yet. I'm not sure we should do that by default, what if the user actually wants to switch between registries in a offline mode. We can probably implement a tidy mode later to clean up unused vendored content.

Reply:
Sounds good!